### ISSUE TITLE
[1.12.x] Crash when used with Better HUD

### ISSUE THREAD

**Issue by Gentle888** [crazysnailboy/AdvancedHUD#1]
Also, somehow, my litemods don't appear on logs, be ensured in that.

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  Inventory Tweaks Coremod (InventoryTweaks-1.63+beta.108.jar)
  BetterFontsCore (TTFR-1.12.1-1.9.1.jar)
Contact their authors BEFORE contacting forge

// Everything's going to plan. No, really, that was supposed to happen.

Time: 9/18/17 3:31 AM
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
	at net.minecraftforge.client.event.RenderGameOverlayEvent.<init>(RenderGameOverlayEvent.java:86)
	at net.minecraftforge.client.event.RenderGameOverlayEvent.<init>(RenderGameOverlayEvent.java:30)
	at net.minecraftforge.client.event.RenderGameOverlayEvent$Pre.<init>(RenderGameOverlayEvent.java:95)
	at net.minecraftforge.client.GuiIngameForge.pre(GuiIngameForge.java:900)
	at net.minecraftforge.client.GuiIngameForge.func_180480_a(GuiIngameForge.java:254)
	at advancedhud.client.GuiAdvancedHUD.func_175180_a(GuiAdvancedHUD.java:56)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1397)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1117)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at java.lang.Thread.getStackTrace(Unknown Source)

-- World Downloader Mod - Core --
Details:
	WDL version: 4.0.1.0
	Minecraft version: Minecraft 1.12.1 (1.12.1/fml,forge/Forge)
	Expected version: 1.12.1
	Protocol version: 338
	Data version: 1241
	File location: ~~ERROR~~ IllegalArgumentException: URI is not hierarchical
Stacktrace:
	at java.lang.Thread.getStackTrace(Unknown Source)
	at net.minecraft.crash.CrashReportCategory.func_85073_a(CrashReportCategory.java:116)
	at net.minecraft.crash.CrashReport.func_85057_a(CrashReport.java:352)
	at wdl.WDL.addInfoToCrash(WDL.java:1469)
	at wdl.WDLHooks.onCrashReportPopulateEnvironment(WDLHooks.java:371)
	at net.minecraft.crash.CrashReport.handler$onCrashReportPopulateEnvironment$zze000(CrashReport.java:514)
	at net.minecraft.crash.CrashReport.func_71504_g(CrashReport.java:144)
	at net.minecraft.crash.CrashReport.<init>(CrashReport.java:54)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:426)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- World Downloader Mod - Extensions --
Details:
	Number loaded: : 2
	Hologram: Id: Hologram
Version: 2.0
Display name: Hologram support
Main author: Pokechu22
Description: 
Provides basic support for disabling holograms.
Main class: wdl.HologramHandler
Containing file: Unknown (java.lang.IllegalArgumentException: URI is not hierarchical)
Implemented interfaces (2)
0: wdl.api.IEntityManager
1: wdl.api.IWDLModDescripted
Superclass: java.lang.Object
Classloader: net.minecraft.launchwrapper.LaunchClassLoader@ba8a1dc (net.minecraft.launchwrapper.LaunchClassLoader)
Annotations (0)

	EntityRealigner: Id: EntityRealigner
Version: 1.0
Display name: Entity realigner
Main author: Pokechu22
Description: 
Realigns entities to their serverside position to deal with entities that drift clientside (for example, boats).
Main class: wdl.EntityRealigner
Containing file: Unknown (java.lang.IllegalArgumentException: URI is not hierarchical)
Implemented interfaces (2)
0: wdl.api.IEntityEditor
1: wdl.api.IWDLModDescripted
Superclass: java.lang.Object
Classloader: net.minecraft.launchwrapper.LaunchClassLoader@ba8a1dc (net.minecraft.launchwrapper.LaunchClassLoader)
Annotations (0)


-- World Downloader Mod - State --
Details:
	minecraft: net.minecraft.client.Minecraft@7aaf0f0a
	worldClient: ~~NULL~~
	networkManager: ~~NULL~~
	thePlayer: ~~NULL~~
	windowContainer: ~~NULL~~
	lastClickedBlock: ~~NULL~~
	lastEntity: ~~NULL~~
	saveHandler: ~~NULL~~
	chunkLoader: ~~NULL~~
	newTileEntities: {}
	newEntities: {}
	newMapDatas: {}
	downloading: false
	isMultiworld: false
	propsFound: false
	startOnChange: false
	overrideLastModifiedCheck: false
	saving: false
	worldLoadingDeferred: false
	worldName: WorldDownloaderERROR
	baseFolderName: WorldDownloaderERROR

-- World Downloader Mod - Base properties --
Details:
	-: empty

-- World Downloader Mod - World properties --
Details:
	-: empty

-- World Downloader Mod - Global properties --
Details:
	TutorialShown: true
	UpdateETag: "8e297e560d1da4c2cd4339c1e45e8b1f"

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['Gentle'/334, l='MpServer', x=1275.97, y=105.68, z=256.22]]
	Chunk stats: MultiplayerChunkCache: 0, 0
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (-200,64,244), Chunk: (at 8,4,4 in -13,15; contains blocks -208,0,240 to -193,255,255), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 15475 game time, 12450 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 1 total; [EntityPlayerSP['Gentle'/334, l='MpServer', x=1275.97, y=105.68, z=256.22]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Integrated singleplayer server

-- System Details --
Details:
	Minecraft Version: 1.12.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_131, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 339164168 bytes (323 MB) / 924999680 bytes (882 MB) up to 2134114304 bytes (2035 MB)
	JVM Flags: 10 total; -XX:HeapDumpPath=ThisTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2048M -Xms256M -Xmn128M -XX:+UseConcMarkSweepGC -XX:-UseAdaptiveSizePolicy -XX:+CMSParallelRemarkEnabled -XX:+ParallelRefProcEnabled -XX:+CMSClassUnloadingEnabled -XX:+UseCMSInitiatingOccupancyOnly
	IntCache: cache: 0, tcache: 0, allocated: 13, tallocated: 95
	FML: MCP 9.41 Powered by Forge 14.22.1.2480 Optifine OptiFine_1.12.1_HD_U_C5 10 mods loaded, 10 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State     | ID              | Version               | Source                            | Signature                                |
	|:--------- |:--------------- |:--------------------- |:--------------------------------- |:---------------------------------------- |
	| UCHIJAAAA | minecraft       | 1.12.1                | minecraft.jar                     | None                                     |
	| UCHIJAAAA | mcp             | 9.19                  | minecraft.jar                     | None                                     |
	| UCHIJAAAA | FML             | 8.0.99.99             | forge-1.12.1-14.22.1.2480.jar     | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | forge           | 14.22.1.2480          | forge-1.12.1-14.22.1.2480.jar     | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | betterfonts     | 1.9.1                 | minecraft.jar                     | None                                     |
	| UCHIJAAAA | advancedhud     | 0.1.10                | AdvancedHUD-1.12-0.1.10.jar       | None                                     |
	| UCHIJAAAA | fw              | 1.6.0                 | FullscreenWindowed-1.12-1.6.0.jar | None                                     |
	| UCHIJAAAA | hud             | 1.3.9                 | hud-1.3.9-1.12.jar.zip            | None                                     |
	| UCHIJAAAA | inventorytweaks | 1.63+beta.108.6ff67c4 | InventoryTweaks-1.63+beta.108.jar | 55d2cd4f5f0961410bf7b91ef6c6bf00a766dcbe |
	| UCHIJAAAA | mousetweaks     | 2.8                   | MouseTweaks-2.8-mc1.12.1.jar      | None                                     |

	Loaded coremods (and transformers): 
Inventory Tweaks Coremod (InventoryTweaks-1.63+beta.108.jar)
  invtweaks.forge.asm.ContainerTransformer
BetterFontsCore (TTFR-1.12.1-1.9.1.jar)
  cubex2.ttfr.BFClassTransformer
	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.5.0 NVIDIA 382.53' Renderer: 'GeForce GTX 960M/PCIe/SSE2'
	Launched Version: 1.12.1
	LWJGL: 2.9.4
	OpenGL: GeForce GTX 960M/PCIe/SSE2 GL version 4.5.0 NVIDIA 382.53, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: Default Improved 1.12.zip
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 8x Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
	OptiFine Version: OptiFine_1.12.1_HD_U_C5
	Render Distance Chunks: 8
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: SEUS-v11.0.zip
	OpenGlVersion: 4.5.0 NVIDIA 382.53
	OpenGlRenderer: GeForce GTX 960M/PCIe/SSE2
	OpenGlVendor: NVIDIA Corporation
	CpuCount: 8
```

**Comment by crazysnailboy** [comment]
This appears to be a compatibility issue with the Better HUD mod. I've a good idea what's causing it - watch this space for a fix :)

**Comment by NekomataGami** [comment]
Looking forward for an update! I really want this mod to work.

**Comment by crazysnailboy** [comment]
I've published v0.1.11 which fixes quite a few bugs. I tested it in with Better HUD and it doesn't crash any more :) Any problems let me know!
